web/meetup-facilitator/src/main.jsx

```diff
@@ -1,13 +1,34 @@
-import React from 'react'
-import ReactDOM from 'react-dom/client'
-import App from './App.jsx'
-import { GoogleOAuthProvider } from '@react-oauth/google'
-import './index.css'
-
-ReactDOM.createRoot(document.getElementById('root')).render(
-  <GoogleOAuthProvider clientId={import.meta.env.VITE_CLIENT_ID}>
-    <React.StrictMode>
-      <App />
-    </React.StrictMode>
-  </GoogleOAuthProvider>
-  )
\ No newline at end of file
+import React, { useEffect } from 'react';
+import ReactDOM from 'react-dom/client';
+import App from './App.jsx';
+import { GoogleOAuthProvider } from '@react-oauth/google';
+import './index.css';
+
+const apiKey ="AIzaSyABQ4Pci5rnkud1UShQlNtU7iGG7waWghU"
+
+const renderApp = () => {
+  ReactDOM.createRoot(document.getElementById('root')).render(
+    <GoogleOAuthProvider clientId={import.meta.env.VITE_CLIENT_ID}>
+      <React.StrictMode>
+        <App />
+      </React.StrictMode>
+    </GoogleOAuthProvider>
+  );
+};
+
+if (apiKey) {
+  // If API key is available, dynamically load Google Maps API
+  const script = document.createElement('script');
+  script.src = `https://maps.googleapis.com/maps/api/js?key=AIzaSyABQ4Pci5rnkud1UShQlNtU7iGG7waWghU&libraries=places`;
+  script.async = true;
+  script.defer = true;
+
+  script.onload = renderApp; // Render the app once the Google Maps API script has loaded
+
+  document.head.appendChild(script);
```

Comment:
I'm a little confused as to why we're using 2 different methods to load the Google Maps API throughout the app. Couldn't we have reused the same method we used for location preferences? Let's leave it for now though for the sake of time.